oshi-core/src/main/java/oshi/software/os/unix/freebsd/FreeBsdOSThread.java

```diff
@@ -0,0 +1,162 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package oshi.software.os.unix.freebsd;
+
+import oshi.software.common.AbstractOSThread;
+import oshi.software.os.OSProcess;
+import oshi.util.ExecutingCommand;
+import oshi.util.ParseUtil;
+
+import java.util.List;
+
+public class FreeBsdOSThread extends AbstractOSThread {
+
+    private int threadId;
+    private String name = "";
+    private OSProcess.State state = OSProcess.State.INVALID;
+    private long minorFaults;
+    private long majorFaults;
+    private long startMemoryAddress;
+    private long contextSwitches;
+    private long kernelTime;
+    private long userTime;
+    private long startTime;
+    private long upTime;
+
+    public FreeBsdOSThread(int processId, String[] split) {
+        super(processId);
+        updateAttributes(split);
+    }
+    @Override
+    public int getThreadId() {
+        return this.threadId;
+    }
+
+    @Override
+    public String getName() {
+        return this.name;
+    }
+
+    @Override
+    public OSProcess.State getState() {
+        return this.state;
+    }
+
+    @Override
+    public long getStartMemoryAddress() {
+        return this.startMemoryAddress;
+    }
+
+    @Override
+    public long getContextSwitches() {
+        return this.contextSwitches;
+    }
+
+    @Override
+    public long getMinorFaults() {
+        return this.minorFaults;
+    }
+
+    @Override
+    public long getMajorFaults() {
+        return this.majorFaults;
+    }
+
+    @Override
+    public long getKernelTime() {
+        return this.kernelTime;
+    }
+
+    @Override
+    public long getUserTime() {
+        return this.userTime;
+    }
+
+    @Override
+    public long getUpTime() {
+        return this.upTime;
+    }
+
+    @Override
+    public long getStartTime() {
+        return this.startTime;
+    }
+
+    @Override
+    public boolean updateAttributes() {
+        String psCommand = "ps -awwxo tdname,lwp,state,etimes,systime,time,tdaddr,nivcsw,nvcsw,majflt -H -p "
```

Comment:
I couldn't find any filter for threads, hence had to us grep command. do you know about any other way?

Reply:
I'm not sure what you're asking here.  If I do `ps -awwxo pid,tdname,lwp,state,etimes,systime,time,tdaddr,nivcsw,nvcsw,majflt -H -p 11` I get only the threads for pid 11.

Reply:
```
$> ps -awwxo pid,tdname,lwp,state,etimes,systime,time,tdaddr,nivcsw,nvcsw,majflt,minflt -H -p 11
PID TDNAME                 LWP STAT ELAPSED SYSTIME      TIME   TDADDR  NIVCSW     NVCSW MAJFLT MINFLT
 11 swi6: task queue    100004 WL    760562 0:00.00   0:00.03 c5f5a880      18        39      0      0
 11 swi6: Giant taskq   100005 WL    760562 0:00.00   0:00.02 c5f5a440       0        74      0      0
 11 swi5: fast taskq    100007 WL    760562 0:00.00   0:00.00 c5f59880       0         0      0      0
 11 swi1: netisr 0      100010 WL    760562 0:00.00  11:41.43 c5f95880  121124  14073708      0      0
 11 swi4: clock (0)     100011 WL    760562 0:00.00  14:52.95 c5f95440 4772170  14241267      0      0
 11 swi4: clock (1)     100012 WL    760562 0:00.00   0:00.00 c5f95000       0         0      0      0
 11 swi3: vm            100013 WL    760562 0:00.00   0:00.00 c5f94880       0         0      0      0
 11 gic0,s19: cesa0     100026 WL    760562 0:00.01   0:01.78 c5fd8440       0    289936      0      0
 11 gic0,s20: cesa1     100027 WL    760562 0:00.00   0:00.00 c5fd8000       0         0      0      0
 11 gic0,s1: spi0       100028 WL    760562 0:00.00   0:00.00 c61b9880       0         0      0      0
 11 swi0: uart uart     100029 WL    760562 0:00.00   0:00.00 c61b9440       0       776      0      0
 11 mpic0: mvneta0      100031 WL    760562 0:00.00   0:00.00 c61b8880       0         0      0      0
 11 mpic0: mvneta1      100034 WL    760562 0:00.00  81:33.78 c61b7880       0 116198736      0      0
 11 gic0,s18: ehci0     100035 WL    760562 0:00.00   0:00.00 c61b7440       0         0      0      0
 11 mpic0: mvneta2      100042 WL    760562 0:00.00 108:48.18 c6340000       0 147196077      0      0
 11 gic0,s26: ahci0     100043 WL    760562 0:00.00   0:22.66 c633b880    8944   3281193      0      0
 11 gic0,s25:-dhci_fdt0 100044 WL    760562 0:00.00   0:00.01 c633b440       0      1504      0      0
 11 gic0,s17: xhci0     100045 WL    760562 0:00.00   0:00.00 c633b000       0         0      0      0
 11 swi1: pf send       100053 WL    760562 0:00.00   0:00.00 c651d440       0        24      0      0
 11 swi1: pfsync        100054 WL    760562 0:00.00   0:00.00 c651d000       0         0      0      0
 11 swi1: netisr 1      100073 WL    760562 0:00.00   0:00.00 c6d08880       0         0      0      0
```

Reply:
Oh, maybe I should look at the source code first to see the context. We're in the thread object!  

No, I don't think there's a switch to filter PS.  But don't pipe `grep` to the OS. Just fetch the `List<String>` and parse the returned string using `contains()` as the Java equivalent.

Reply:
You might also consider using `procstat` for some info as well:
```
> procstat -t 11
  PID    TID COMM                TDNAME              CPU  PRI STATE   WCHAN
   11 100004 intr                swi6: task queue     -1   48 wait    -
   11 100005 intr                swi6: Giant taskq    -1   48 wait    -
   11 100007 intr                swi5: fast taskq     -1   44 wait    -
   11 100010 intr                swi1: netisr 0       -1   28 wait    -
   11 100011 intr                swi4: clock (0)      -1   40 wait    -
   11 100012 intr                swi4: clock (1)      -1   40 wait    -
   11 100013 intr                swi3: vm             -1   36 wait    -
   11 100026 intr                gic0,s19: cesa0      -1    8 wait    -
   11 100027 intr                gic0,s20: cesa1      -1    8 wait    -
   11 100028 intr                gic0,s1: spi0        -1   20 wait    -
   11 100029 intr                swi0: uart uart      -1   24 wait    -
   11 100031 intr                mpic0: mvneta0       -1    8 wait    -
   11 100034 intr                mpic0: mvneta1       -1    8 wait    -
   11 100035 intr                gic0,s18: ehci0      -1   12 wait    -
   11 100042 intr                mpic0: mvneta2       -1    8 wait    -
   11 100043 intr                gic0,s26: ahci0      -1   12 wait    -
   11 100044 intr                gic0,s25:-dhci_fdt0  -1   20 wait    -
   11 100045 intr                gic0,s17: xhci0      -1   12 wait    -
   11 100053 intr                swi1: pf send        -1   28 wait    -
   11 100054 intr                swi1: pfsync         -1   28 wait    -
   11 100073 intr                swi1: netisr 1       -1   28 wait    -
```

Reply:
(`procstat` has the priority, not sure if that's available on `ps`?)

Reply:
OK, you can add `pri` to the `ps` string as well but it's interesting, it shows up as a negative number offset by 100, e.g., TID 100004 has pri 48 in `procstat` and -52 on `ps`.

Reply:
Added priority. `ps` has priority and a value of `actual priority - 100` . Checked with `procstat` and it matches.